### ISSUE TITLE
Desktop - Black Screen - Windows

### ISSUE THREAD

**Issue by Kurtscob** [Ylianst/MeshCentral#315]
Hi,

Recently, my connections using meshcentral 2 on Windows 10 computers have resulted in the desktop connection remaining as a black screen instead of displaying the remote desktop of the connected computer.  I am still able to send that computer commands like reset under actions.

I use the x64 windows.exe for backgrounds and interactive to connect. Ihave noticed that when I do the update/install of the file on the remote computer that it no longer creates a 4 Kb .msh file.  I am not sure if that is part of the issue.

Can anyone tell me why I can no longer see the desktop of the computers that I connect to

Thank you

K

**Comment by FilGov** [comment]
Same issue.

**Comment by merlinweb** [comment]
I can confirm the same problem. Sometimes, disconnecting and reconnecting after a couple of seconds could solve, sometimes is black again.
other times, desktop composing is extremely slow, even in gigabit lan, with little squares filling remote desktop following mouse movement. Disconnecting and reconnecting seems speed up refreshing

Hope this helps

Cheers

Merlinweb

**Comment by FilGov** [comment]
Same issues.

**Comment by krayon007** [comment]
Can you guys describe your network topology? I'm having a hard time reproducing this issue.

Based on what I'm hearing, it sounds like a connection is hanging around after disconnecting, so when another client connects, it gets multiplexed with it. The stake connection is then screwing up the flow control of the mutliplezed stream....

I'm curious if you guys are using a reverse proxy, perhaps that's where I should look to make sure the proxy is detecting/propagating the connection close. I could be totally off base tho.

**Comment by FilGov** [comment]
Sure: home computer - home router - meshcentral.com - work router - work computers. Not (knowingly) using any reverse proxy.

**Comment by merlinweb** [comment]
Hi Bryan,
I am in LAN... Meshcentral is on a VM (win serv 2016) On an HPE Blade Enclosure, connected to an HPE 5412z core switch, linked to enclosure by a 10G DAC (or fiber) cable
Clients I'm connecting are in the same LAN, all gigabit certified with CAT6 cables 

NO proxy / reverse proxy

**Comment by krayon007** [comment]
One more question... What's the state of the machines you are trying to connect to? Is a user logged in, but the screen locked? Is the screen unlocked? Or is it at the login screen with no users logged in?

**Comment by merlinweb** [comment]
User is logged and working
But the issue happens in all three behaviour you have written
Today, I get black screen on a connected user (in wifi), after three cycles of disconnect and connect desktop appeared correctly

**Comment by FilGov** [comment]
User is logged in and screen (read session) is locked and user is logged in and screen (read session) is unlocked.

**Comment by wcrlewis** [comment]
I have the same issue and was here to report.  The problem started around two weeks and a day ago.  If the systems are disconnected and reconnected multiple times sometimes the connection is functional.  Some systems it is nearly impossible.

**Comment by wcrlewis** [comment]
The issue I experience is lack of connectivity for video, mouse and keyboard. 'Actions' do function. In the last two weeks systems often now require half a dozen attempts and some including one I am working with now almost never connects (50+ tries - A restart and immediate connection frequently succeeds).

**Comment by deajan** [comment]
Can confirm this with MeshCentral 2.0-beta2 (using meshcentral.com test server).
I also can confirm that meanwhile, I did not have any connection problems (since my test client computer was also available via SimpleHelp at the same time).

**Comment by GusevVictor** [comment]
Dear all, I have the same problem with version 0.4.9-o. Please fix that. Please please please.

Mesh agent is v 0.2.1.3

**Comment by GusevVictor** [comment]
I suppose there is problem with Apache proxy. I have tune it and wait for that problem again.

**Comment by uldiseihenbergs** [comment]
@GusevVictor any news from you on this issue? Are you fixed then yourself?
i'm asking cos i have something similar black screen but only after i enter user/password. then after Disconnect/Connect screen shows again..

**Comment by FilGov** [comment]
> @GusevVictor any news from you on this issue? Are you fixed then yourself?
> i'm asking cos i have something similar black screen but only after i enter user/password. then after Disconnect/Connect screen shows again..

Same.

**Comment by krayon007** [comment]
I have seen the issue where sometimes when you enter username password, it goes black, and you have to disconnect and reconnect. What happens in that scenario, is that on the logon screen, the child process is root. When you logon, that root process no longer has sufficient permission to scrape the screen, and the process exits. The parent is supposed to trap that, and respawn the child as the logged in user... I've observed sometimes when you logon, the child process doesn't exit, it just hangs....  I'll see if I can do something about this.... If I encounter the other issues (which I haven't as of yet), I'll look into that as well.

**Comment by GusevVictor** [comment]
@krayon007 , please fix it if you can. It is very bad situation for now. Thank, thank for your work.

**Comment by GusevVictor** [comment]
> 
> 
> > @GusevVictor any news from you on this issue? Are you fixed then yourself?
> > i'm asking cos i have something similar black screen but only after i enter user/password. then after Disconnect/Connect screen shows again..
> 
> Same.

Same.

**Comment by GusevVictor** [comment]
> 
> 
> @GusevVictor any news from you on this issue? Are you fixed then yourself?
> i'm asking cos i have something similar black screen but only after i enter user/password. then after Disconnect/Connect screen shows again..

No news I still have the problem

**Comment by FilGov** [comment]
Me too. Using meshcentral.com.

**Comment by GusevVictor** [comment]
> 
> 
> I have seen the issue where sometimes when you enter username password, it goes black, and you have to disconnect and reconnect. What happens in that scenario, is that on the logon screen, the child process is root. When you logon, that root process no longer has sufficient permission to scrape the screen, and the process exits. The parent is supposed to trap that, and respawn the child as the logged in user... I've observed sometimes when you logon, the child process doesn't exit, it just hangs.... I'll see if I can do something about this.... If I encounter the other issues (which I haven't as of yet), I'll look into that as well.

Dear @krayon007 there is an anothter situation for meet the problem.

**For example:** imagine that you have an non good Internet connection to web mesh from home. When you internet hangs you will disconnect and after disconnect you **sometimes** will see a black screen. In a real local monitor you will see two connected **same remote user** to desktop. For fix that you need to restart mesh agent on a remote desktop and reconnect again from web. It is very bad situation. **If you can fix that it will be amazing. Thank.**

**Comment by datapharmer** [comment]
We are seeing this issue as well as computers upgrade to the new client via meshcentral.com. Login to pc works but login to view screen with an active desktop user does not.

**Comment by datapharmer** [comment]
Duplicate of #1166

**Comment by datapharmer** [comment]
Ok, so we dug in a bit more and confirmed that if just "Desktop Toolbar" is selected then the connection works as expected, but if "Desktop Prompt+Toolbar" is selected for User Consent then after consent is given it remains with a black screen. Task manager also shows another instance of meshagent being spawned, so it is as if something is hanging after permission is given.  We are on Windows 10 version 1909 on dell latitude and optiplex and this is an install from meshcentral.com.

**Comment by krayon007** [comment]
It looks like this issue was fixed in a newer version of the server than is currently on meshcentral.com. Ylian said he would update the server.

**Comment by deajan** [comment]
I can confirm that with v 0.5.77, having two agents running at the same time produces a black screen.
I ran into this issue while running meshagent in "connect" mode on a machine without realizing it already had a meshagent installed as service.

While the black screen prevented to do anything, I still could login in terminal mode, and use Windows `tasklist| findstr meshagent` then `taskkill /PID <meshagentPID> /F` to get rid of the second agent.
This allowed me to "reclaim" a non black screen on that computer.

**Comment by FilGov** [comment]
I ran into this issue on meshcentral.com with meshagent only installed as a service.

**Comment by datapharmer** [comment]
It looks like when permission is required it spawns a second meshagent instance which may be part of the issue, but in any case as of now this is still happening on meshcentral.com

**Comment by deajan** [comment]
Just found another way to trigger the black screen with meshcentral.
Here's how I got the black screen:
- Install agent
- Lock desktop (CTRL+L) or CTRL+ALT+DEL then Lock
- Disconnect from desktop
- Reconnect to desktop

Unfortunately, I've tried to reproduce the same scenario, but second time I did not get a black screen.
I tried to reproduce the issue another time, only to get the black screen when I locked the desktop.
So in the end I don't have a 100% success rate in reproducing the issue.

I guess there may be multiple black screen issues in meshcentral, of which at least one maybe related to privilege isolation between display and services.

Are there any server side / client side logs that could help diagnose ? Any debug mode I can set to help ?

**Comment by tootai** [comment]
Hello, some news about this blackscreen problem ?

We use meshcentral 0.8.0 and face this problem with our test Windows7-64 WS in a kvm VM. This computer has RDP activate which is working well. Funny side of our test is that we connect to the computer, launch meshagent64 in connected mode, stay connected eg RDP session up, and connect from meshcentral: we see 2 sessions, the RDP one and a Console/connected one. Using the latest, we are able to connect.

Now we restart meshagent from RDP session, disconnect this session and try to connect from meshcentral, black screen. Agent is connected, console react, if we reconnect with RDP everyting is fine and meshagent64 running.

Sometimes after a while, session is stil showed as connected but in th events log we see Desktop session "xyz"ended: from now no more possibility to access the remote desktop (tab disappear as well as terminal one) only console one is present. If we open the RDP session, everything is still fine.

Could it be possible to add a future like restart remote agent from meshcentral ?

Network setup is office were meshcentral lies in a VM Debian/Buster <> Internet fiber channel 1Gbps/300 Mbps <> dedicated server@Hetzner <> VM W7 

Daniel

**Comment by krayon007** [comment]
> Now we restart meshagent from RDP session, disconnect this session and try to connect from meshcentral, black screen. Agent is connected, console react, if we reconnect with RDP everyting is fine and meshagent64 running.

If you are running the agent from the RDP session, this is a Windows Permission issue... When you disconnect the RDP session, windows goes to the login screen. If the agent is running from a user session (RDP), it does NOT have sufficient permission to scrape the login screen. The agent needs to be running as Local System to do that.

**Comment by krayon007** [comment]
> Could it be possible to add a future like restart remote agent from meshcentral ?

This feature is already there... You can use the following command from the console tab
`service restart`
The only caveat, is that this only works if your agent is running as a service.

**Comment by tootai** [comment]
Le 02/04/2021 à 19:32, Bryan Roe a écrit :

>     Now we restart meshagent from RDP session, disconnect this session
>     and try to connect from meshcentral, black screen. Agent is
>     connected, console react, if we reconnect with RDP everyting is
>     fine and meshagent64 running.
>
> If you are running the agent from the RDP session, this is a Windows 
> Permission issue... When you disconnect the RDP session, windows goes 
> to the login screen. If the agent is running from a user session 
> (RDP), it does NOT have sufficient permission to scrape the login 
> screen. The agent needs to be running as Local System to do that.
>
ah ha, didn't think about this. We tried from a physical Windows PC and 
effectively didn't face the black screen.

Thanks for the tip.

-- 
Daniel

**Comment by tootai** [comment]
Le 02/04/2021 à 19:33, Bryan Roe a écrit :

>     Could it be possible to add a future like restart remote agent
>     from meshcentral ?
>
> This feature is already there... You can use the following command 
> from the console tab
> |service restart|
>
|Perfect|
>
> The only caveat, is that this only works if your agent is running as a 
> service.
>
Which seems obvious...

Thanks.

-- 
Daniel

**Comment by yuling3086** [comment]
Hi 

I am using meshcentral version  0.6.97,I try to switch user or user sign in the Window User Account of the remote PC during remote access.We face some issue where the remote access screen will freeze and not able to remote control.

User will have to Stop Remote Access and Exit then reconnect the Remote Access again to take control.

**Comment by Extarys** [comment]
Latest version. Connecting to Windows 10 worked, then blank screen out of nowhere. Tried disconnecting and reconnecting, and same thing. I had to ask the person to disconnect/close the agent and restart it. It happens at least once a week per computer.

No logs I could find server side either. Quite strange. For comparison, I never had this issue with RustDesk. I'll try and see if I can find some logs somewhere.

**Comment by dinger1986** [comment]
> EDIT: nvm, I see two new versions after the one I'm using - my mistake. I'll update and report back.
> 
> Latest version. Connecting to Windows 10 worked, then blank screen out of nowhere. Tried disconnecting and reconnecting, and same thing. I had to ask the person to disconnect/close the agent and restart it. It happens at least once a week per computer.
> 
> No logs I could find server side either. Quite strange. For comparison, I never had this issue with RustDesk. I'll try and see if I can find some logs somewhere.

are you using firefox?

**Comment by GusevVictor** [comment]
> > EDIT: nvm, I see two new versions after the one I'm using - my mistake. I'll update and report back.
> > Latest version. Connecting to Windows 10 worked, then blank screen out of nowhere. Tried disconnecting and reconnecting, and same thing. I had to ask the person to disconnect/close the agent and restart it. It happens at least once a week per computer.
> > No logs I could find server side either. Quite strange. For comparison, I never had this issue with RustDesk. I'll try and see if I can find some logs somewhere.
> 
> are you using firefox?

I'm sorry, that question is not for me but I use firefox and I have the problem from time to time

**Comment by dinger1986** [comment]
Chrome seems to work better fyi

**Comment by Extarys** [comment]
@dinger1986 Ok I'll try with ungoogled chromium and see if it helps ;)

But it's strange that the remote pc's agent needs to be disconnected then reconnected for it to work with FF. Sounds like Firefox might trigger something inside the agent that breaks future connections (or something like that)

**Comment by dinger1986** [comment]
Possibly.

It's worth a shot anyway

**Comment by Extarys** [comment]
I have more black screen with chrome than with firefox. The user need to close and re-open the agent each time it happens. :/
